Automattic/woocommerce-subscriptions-core
# Update deprecated WooCommerce Blocks hook

## Description


This change is needed because recently WooCommerce Blocks graduated some of its experimental hooks (https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5017 and https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5014) to stable, so the `__experimental` prefix was dropped. This PR updates the use of those hooks to the stable version.

Without this PR, deprecation warnings are shown in the Checkout Block.


This code can break if the user does not update their version of WooCommerce Blocks to the latest version that includes the graduated hooks.

Tips/tricks on how to mitigate this would be much appreciated, because I'm really unsure on how we can cater for this situation.



## How to test this PR

Place an order that includes subscription items and ensure the Checkout page does not show any warnings. Ensure the order in WP Admin is correct and has all the expected information.

## Product impact


^ I'm not 100% confident on the above, whether WC Payments would call these methods, I think it won't but could someone who knows more 👍🏼  this one?
